modules/alloy/src/main/ts/ephox/alloy/api/ui/HtmlSelect.ts

```diff
@@ -40,10 +40,18 @@ const factory: SingleSketchFactory<HtmlSelectDetail, HtmlSelectSpec> = (detail,
               return Value.get(select.element);
             },
             setValue: (select, newValue) => {
+              const firstOption = detail.options[0];
```

Comment:
What do you mean?

Reply:
Can you create a HtmlSelect component with zero options and call setValue on it then this will be undefined and accessing properties on it will throw exceptions. But maybe schemas etc prevents that from happening.

Reply:
I do believe the schemas do mean we would have an empty array here. I have tested with no children before, and it doesn't crash here. It'd crash later when we tried to use the element we got from the array ( since it was nullable ), but that's what the `Type.isNonNullable(firstOption)` is for.

Reply:
If the schemas doesn't fix it the typing is also very wrong.

Reply:
You could use Arr.head to use an option and avoid nullish behaviour.

Reply:
We'd still have to do the same test in the same place, we'd just be wrapping it in an optional. I don't think it's really any improvement to speed or readability.

Reply:
I think this is a general problem people reach for null checks over options to safe guard the code. I think you make a mentality shift towards options if you are exposed to a language that doesn't have null. The same goes for exceptions vs Results, mutable state vs cells with immutable data inside. There is a reason modern languages like say Rust doesn't have null and instead has these structures. One could argue that the ts compiler in strict mode now forces you to do null checks but it's a weak safety and it can easily be tricked in this case it's fine but if you do structural typing then it not. Wrapping everything in a some/none construct forces you to deal with both cases and you can never forget to deal with them since you need to unwrap to get to the real value. They also compose and have higher order functions on them. I don't think creating an option here is a performance problem since you would have to have a massive amounts of select components for that to be a bottle neck and it likely other hot spots that would be more problematic. Sure I occasionally use null/undefined checks but that is normally in code written prior to switching to these fp concepts.

Reply:
I am not saying optionals are bad all the time, just that in this situation I don't see it being an improvement but rather make it less clear to read. But if you feel so strongly about it, I'll change it.